examples/images/src/main.rs

```diff
@@ -20,18 +20,27 @@ fn main() -> Result<(), eframe::Error> {
     )
 }
 
-#[derive(Default)]
-struct MyApp {}
+struct MyApp {
+    gif: egui::Image<'static>,
+}
+
+impl Default for MyApp {
+    fn default() -> Self {
+        Self {
+            gif: egui::Image::new_animation(egui_extras::include_gif!("ferris.gif")),
+        }
+    }
+}
 
 impl eframe::App for MyApp {
     fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
         egui::CentralPanel::default().show(ctx, |ui| {
             egui::ScrollArea::both().show(ui, |ui| {
+                ui.add(self.gif.clone());
```

Comment:
it could be changed to something like this:

```rs
macro_rules! include_gif {
    ($path:expr $(,)?) => {
        $crate::ImageSource::Bytes {
            uri: ::std::borrow::Cow::Borrowed(concat!("gif_bytes://", $path)),
            bytes: $crate::load::Bytes::Static(include_bytes!($path)),
        }
    };
}
```
the new function will check if the uri starts with gif_bytes & look it up/call gif_to_sources

The problem is how to cache the gif.  @emilk do you have an idea how to cache the decoded gif?

Something like this would be needed, but how. static could be used, but it would require unsafe code. `fn get_cached(uri: &str) -> Option<AnimatedImageSource<'a>>;`